coordinator/coordinator.go

```diff
@@ -590,6 +590,20 @@ func (c *Coordinator) runRegisterSourceJob(resID metadata.ResourceID, schedule s
 			c.Logger.Errorf("could not close offline store: %v", err)
 		}
 	}(sourceStore)
+	status := source.Status()
+	if status == metadata.READY {
```

Comment:
I copied this from label and ts. Seems weird though no? Feel like we shouldn't return an error if it's already completed.